README.md

```diff
@@ -43,6 +43,12 @@ if (!map.has(key)) {
 map.get(key).doThing();
 ```
 
+With this proposal:
+
+```js
+map.upsert(key, () => {}, () => value).doThing()
```

Comment:
the update function needs to be `undefined` or an explicit identity function like `v => v`